iothub/device/src/IotHubBaseClient.cs

```diff
@@ -388,14 +390,15 @@ public async Task<Twin> GetTwinAsync(CancellationToken cancellationToken = defau
         /// </summary>
         /// <param name="reportedProperties">Reported properties to push</param>
         /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+        /// <returns>The new version of the updated twin if the update was successful.</returns>
         /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
-        public async Task UpdateReportedPropertiesAsync(TwinCollection reportedProperties, CancellationToken cancellationToken = default)
+        public async Task<int> UpdateReportedPropertiesAsync(TwinCollection reportedProperties, CancellationToken cancellationToken = default)
```

Comment:
Our pnp work chose to expose this as a long, although I forget why. It is possible the Version in a Twin is saved as a long.
Also, even though we chose to expose this with the request Id for pnp, I think the Version alone is what is going to be useful to the user.

Reply:
I'll try to dig some more into if this value should be a long or an int

Reply:
Currently, the user doesn't get to pick the request id when they make this request, so I don't think there is any value in returning the request id to the user here. Even if we did allow them to pick the request id, the service returns the same request id in its response message so there is no reason to expose it then either.